sys/net/gnrc/network_layer/sixlowpan/frag/sfr/gnrc_sixlowpan_frag_sfr.c

```diff
@@ -0,0 +1,1546 @@
+    uint16_t datagram_size = sixlowpan_sfr_rfrag_get_offset(hdr);
+    uint8_t fragment_size = sixlowpan_sfr_rfrag_get_frag_size(hdr);
+    uint8_t *payload;
+
+    if ((datagram_size == 0) && (fragment_size == 0)) {
+        bool release_pkt = true;
+
+        DEBUG("6lo sfr: Abort for datagram (%s, %u) received\n",
+              gnrc_netif_addr_to_str(gnrc_netif_hdr_get_src_addr(netif_hdr),
+                                     netif_hdr->src_l2addr_len, addr_str),
+              hdr->base.tag);
+        if ((entry->entry.vrb = gnrc_sixlowpan_frag_vrb_get(
+                gnrc_netif_hdr_get_src_addr(netif_hdr),
+                netif_hdr->src_l2addr_len, hdr->base.tag)) != NULL) {
+            entry->type = _VRB;
+            _forward_rfrag(pkt, entry, 0, page);
+            gnrc_sixlowpan_frag_vrb_rm(entry->entry.vrb);
+            release_pkt = false;
+        }
+        if ((entry->entry.rb = gnrc_sixlowpan_frag_rb_get_by_datagram(
+                netif_hdr, hdr->base.tag)) != NULL) {
+            entry->type = _RB;
+            _abort_rb(pkt, entry, netif_hdr, sixlowpan_sfr_rfrag_ack_req(hdr));
+            release_pkt = false;
+        }
+        if (release_pkt) {
+            DEBUG("6lo sfr: received abort for unknown datagram\n");
+            /* neither VRB or RB exists => release packet */
+            gnrc_pktbuf_release(pkt);
+        }
+        return;
+    }
+    DEBUG("6lo sfr: First fragment (%s, %u) received\n",
+          gnrc_netif_addr_to_str(gnrc_netif_hdr_get_src_addr(netif_hdr),
+                                 netif_hdr->src_l2addr_len, addr_str),
+          hdr->base.tag);
+
+    payload = (uint8_t *)(hdr + 1);
+    if (payload[0] == SIXLOWPAN_UNCOMP) {
+        _forward_uncomp(netif_hdr, pkt, page, entry, payload + 1);
+    }
+    else {
+        if (IS_USED(MODULE_GNRC_SIXLOWPAN_IPHC) &&
+            sixlowpan_iphc_is(payload)) {
+            _try_reassembly(netif_hdr, pkt, 0, entry, page);
+            return;
+        }
+        DEBUG("6lo sfr: unable to parse next dispatch for forwarding "
+              "information. Abort\n");
+        _abort_rb(pkt, entry, netif_hdr, false);
+        return;
+    }
+}
+
+static void _handle_nth_rfrag(gnrc_netif_hdr_t *netif_hdr, gnrc_pktsnip_t *pkt,
+                              unsigned page, _generic_rb_entry_t *entry)
+{
+    sixlowpan_sfr_rfrag_t *hdr = pkt->data;
+    uint16_t offset = sixlowpan_sfr_rfrag_get_offset(hdr);
+
+    DEBUG("6lo sfr: Subsequent fragment (%s, %u) received\n",
+          gnrc_netif_addr_to_str(gnrc_netif_hdr_get_src_addr(netif_hdr),
+                                 netif_hdr->src_l2addr_len, addr_str),
+          hdr->base.tag);
+    if (gnrc_sixlowpan_frag_rb_exists(netif_hdr, hdr->base.tag)) {
+        DEBUG("6lo sfr: I am destination endpoint => adding to reassembly "
+              "buffer\n");
+        _try_reassembly(netif_hdr, pkt, offset, entry, page);
+    }
+    else if ((entry->entry.vrb = gnrc_sixlowpan_frag_vrb_get(
```

Comment:
a short comment to explain the cases would be appreciated :)

Reply:
Added